com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/json/references/JsonDocumentManager.java

```diff
@@ -91,11 +91,6 @@ public JsonNode getDocument(URL url) {
     }
 
     public JsonNode getDocument(URI uri) {
-        final IFile file = getFile(uri);
-        if (file == null || !file.exists()) {
-            return null;
-        }
-
         try {
             return getDocument(uri.toURL());
```

Comment:
We are passing an entire URL to `JsonDocumentManager.getDocument(URL)` which includes both the `file` and `ref` parts. `JsonDocumentManager.getDocument(URL)` has a table whose keys are `URL`s in a form we are passing the here. As a result, for two different refs from the **same** external model will load the model twice. For example, for the `beamup_path_estimates_price.yaml` from the Multi-file Beamup (v3) example the `documents` map will contain a different JsonNode model for each parameter:
```
{<...>/Multi-file%20Beamup%20(v3)/models/components/beamup_parameters.yaml#/components/parameters/start_longitude={"openapi":"3.0.0", <...>}, 

<...>/Multi-file%20Beamup%20(v3)/models/components/beamup_parameters.yaml#/components/parameters/end_longitude={"openapi":"3.0.0",<...>}
}
```
But it should be a single entry for `<...>/Multi-file%20Beamup%20(v3)/models/components/beamup_parameters.yaml`! 
We should fix `JsonDocumentManager.getDocument(URL)`.